From f2d1163b3c965b8b61f98460ff68d6cd94aae1a0 Mon Sep 17 00:00:00 2001
From: "Carsten Haitzler (Rasterman)" <raster@rasterman.com>
Date: Sun, 13 Jun 2021 12:15:18 +0100
Subject: [PATCH 2/2] alpm - fix missing handling of hookdir in pacman.conf

The pacman.conf parser didnt handle HookDir at all. This fixes that
and handles it.
---
 backends/alpm/pk-alpm-config.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/backends/alpm/pk-alpm-config.c b/backends/alpm/pk-alpm-config.c
index 3fbb60c18..3bce61a52 100644
--- a/backends/alpm/pk-alpm-config.c
+++ b/backends/alpm/pk-alpm-config.c
@@ -47,7 +47,7 @@ typedef struct
 
 	 alpm_list_t		*cachedirs, *holdpkgs, *ignoregroups,
 				*ignorepkgs, *localfilesiglevels, *noextracts,
-				*noupgrades, *remotefilesiglevels;
+				*noupgrades, *remotefilesiglevels, *hookdirs;
 
 	 alpm_list_t		*sections;
 	 GRegex			*xrepo, *xarch;
@@ -106,6 +106,7 @@ pk_alpm_config_free (PkAlpmConfig *config)
 	FREELIST (config->noextracts);
 	FREELIST (config->noupgrades);
 	FREELIST (config->remotefilesiglevels);
+	FREELIST (config->hookdirs);
 
 	alpm_list_free_inner (config->sections, pk_alpm_config_section_free);
 	alpm_list_free (config->sections);
@@ -360,6 +361,7 @@ static const PkAlpmConfigList pk_alpm_config_list_options[] = {
 	{ "NoUpgrade", G_STRUCT_OFFSET (PkAlpmConfig, noupgrades) },
 	{ "RemoteFileSigLevel", G_STRUCT_OFFSET (PkAlpmConfig,
 						 remotefilesiglevels) },
+	{ "HookDir", G_STRUCT_OFFSET (PkAlpmConfig, hookdirs) },
 	{ NULL, 0 }
 };
 
@@ -997,6 +999,10 @@ pk_alpm_config_configure_alpm (PkBackend *backend, PkAlpmConfig *config, GError
 	alpm_option_set_noupgrades (handle, config->noupgrades);
 	config->noupgrades = NULL;
 
+	/* alpm takes ownership */
+	alpm_option_set_noupgrades (handle, config->hookdirs);
+	config->hookdirs = NULL;
+
 	pk_alpm_config_configure_repos (backend, config, handle, error);
 
 	return handle;
-- 
2.31.1

